fapolicy_analyzer/ui/rules/rules_difference_dialog.py

```diff
@@ -22,6 +22,17 @@
 from gi.repository import Gtk  # isort: skip
 
 
+def filter_rule_diff(previous_system, current_system):
+    diffs = rules_difference(previous_system, current_system).split("\n")
+    for d in diffs:
+        if (
+           (d.startswith("+") and d.split("+")[-1] + "\n" in previous_system.rules_text())
+           or (d.startswith("-") and d.split("-")[-1].split("\n")[0] in current_system.rules_text())
+           ):
+            diffs.remove(d)
+    return diffs
```

Comment:
I think you can add part of your if check there as well and return a tuple of `(adds, dels)`  instead single list. Something like 

```python
adds = [d for d in diffs if d.startswith("+") and f"{d.split("+")[-1]}\n" in previous_system.rules_text()]
dels = [d for d in diffs if d.startswith("-") and d.split("-")[-1].split("/n")[0] in current_system.rules_text()]
return (adds, dels)
```

Then you can call it like so `adds, dels = filter_rule_diff(previous_system, current_system)`

Reply:
This check is in place to handle instances where there would be an addition and a deletion detected when removing or adding a new line to the rules file where the only difference is a newline. 

The `RulesDifferenceDialog` and the `ConfirmDeploymentDialog` both use the returned list slightly differently where a list of all additions and all deletions wouldn't work because the `RulesDifferenceDialog` cares about which `system` the addition or deletion came from.